### PR TITLE
Fixed phpdoc

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

File: Admin/CommentAdmin.php [hunk 1]
```diff
@@ -146,9 +146,9 @@ public function postUpdate($object)
     }
 
     /**
-     * @param ManagerInterface $commentManager
+     * @param CommentManagerInterface $commentManager
      */
-    public function setCommentManager(ManagerInterface $commentManager)
+    public function setCommentManager(CommentManagerInterface $commentManager)
```
Comment (Soullivaneuh) [inline Admin/CommentAdmin.php]: I think this is BC break. BTW, why the interface is changed?

Comment (core23) [inline Admin/CommentAdmin.php]: Because we call the ``updateCommentsCount`` method in https://github.com/core23/SonataNewsBundle/blob/phpdoc/Admin/CommentAdmin.php#L121

Same reason for constructor change on ``RecentCommentsBlockService`` and ``RecentPostsBlockService``.

Comment (core23) [inline Admin/CommentAdmin.php]: @Soullivaneuh Okay, or should I split this in two seperate PRs: phpdoc and interface changes?

Comment (Soullivaneuh) [inline Admin/CommentAdmin.php]: If possible yes.

And in any case the BC break need to be handled.

**Comment by core23** [comment]
Now an error is thrown if you pass an invalid ManagerInterface to the corresponding class.

File: Admin/CommentAdmin.php [hunk 2]
```diff
@@ -150,6 +151,9 @@ public function postUpdate($object)
      */
     public function setCommentManager(ManagerInterface $commentManager)
     {
+        if (!$commentManager instanceof CommentManagerInterface) {
+            trigger_error('setCommentManager() expected a CommentManagerInterface.', E_USER_WARNING);
```
Comment (Soullivaneuh) [inline Admin/CommentAdmin.php]: Should be a deprecation error with a proper message.

File: Admin/CommentAdmin.php [hunk 3]
```diff
@@ -150,6 +151,9 @@ public function postUpdate($object)
...
+        }
         $this->commentManager = $commentManager;
```
Comment (Soullivaneuh) [inline Admin/CommentAdmin.php]: Please add a new line.

**Comment by core23** [comment]
Added a version information to the trigger_error message

File: Admin/CommentAdmin.php [hunk 4]
```diff
@@ -150,6 +151,10 @@ public function postUpdate($object)
      */
     public function setCommentManager(ManagerInterface $commentManager)
     {
+        if (!$commentManager instanceof CommentManagerInterface) {
+            trigger_error('The '.__METHOD__.' method expected a CommentManagerInterface since version 2.4', E_USER_DEPRECATED);
```
Comment (OskarStark) [inline Admin/CommentAdmin.php]: expect/s instead of expected?

Comment (Soullivaneuh) [inline Admin/CommentAdmin.php]: Please follow this message convention: https://github.com/symfony/symfony/blob/eca3e37225614ab49feb8cd1763ea1ebee76edd7/src/Symfony/Component/OptionsResolver/OptionsResolver.php#L479

And use FQN for classes.

**Comment by OskarStark** [comment]
Can you please finally review this PR @Soullivaneuh

File: Block/RecentCommentsBlockService.php [hunk 5]
```diff
@@ -72,9 +82,9 @@ public function execute(BlockContextInterface $blockContext, Response $response
     /**
      * {@inheritdoc}
      */
-    public function validateBlock(ErrorElement $errorElement, BlockInterface $block)
```
Comment (Soullivaneuh) [inline Block/RecentCommentsBlockService.php]: Why this is removed?

Comment (core23) [inline Block/RecentCommentsBlockService.php]: Beause it' inherited from https://github.com/sonata-project/SonataBlockBundle/blob/master/Block/BaseBlockService.php#L198

Comment (Soullivaneuh) [inline Block/RecentCommentsBlockService.php]: Indeed. But this is not related to the PR.

**Comment by core23** [comment]
Split this request in three commits #265 #267 #266

**Comment by core23** [comment]
@Soullivaneuh Ready to merge

**Comment by Soullivaneuh** [comment]
Could you please rebase? Missing checks.

**Comment by core23** [comment]
Rebased @Soullivaneuh

**Comment by OskarStark** [comment]
Thank you @core23
